src/NodaTime.Demo/ZonedDateTimeDemo.cs

```diff
@@ -39,5 +40,183 @@ public void SkippedLocalDateTime()
+
+            Duration duration = nearEndOfDay - startOfDay;
+            Assert.AreEqual(Duration.FromHours(25) - Duration.FromSeconds(1), duration);
+
+            Assert.AreNotEqual(duration.TotalTicks, nearEndOfDay.TickOfDay);
+        }
+
+        [Test]
+        public void AddDurationAndIsDaylightSavingTime()
+        {
+            // This is the date when DST ended at 2am in the given time zone.
+            var dt = new LocalDateTime(2017, 10, 29, 1, 45, 0);
+
+            DateTimeZone dublin = DateTimeZoneProviders.Tzdb["Europe/Dublin"];
+
+            ZonedDateTime beforeTransition = new ZonedDateTime(dt, dublin, Offset.FromHours(1));
+
+            Assert.AreEqual(true, Snippet.For(beforeTransition.IsDaylightSavingTime()));
+            var result = Snippet.For(ZonedDateTime.Add(beforeTransition, Duration.FromHours(1)));
+            Assert.AreEqual(new LocalDate(2017, 10, 29), result.Date);
+            Assert.AreEqual(new LocalTime(1, 45, 0), result.TimeOfDay);
+            Assert.AreEqual(Offset.FromHours(0), result.Offset);
+            Assert.AreEqual(false, result.IsDaylightSavingTime());
+
+            var result2 = Snippet.For(beforeTransition + Duration.FromHours(1));
+            Assert.AreEqual(result, result2);
+
+            ZonedDateTime afterTransition = new ZonedDateTime(dt, dublin, Offset.FromHours(0));
+            result = ZonedDateTime.Add(afterTransition, Duration.FromDays(1));
+            Assert.AreEqual(new LocalTime(1, 45, 0), result.TimeOfDay);
+            Assert.AreEqual(new LocalDate(2017, 10, 30), result.Date);
+        }
+
+        [Test]
+        public void PlusDuration()
+        {
+            // This is the date when DST ended at 2am in the given time zone.
+            var dt = new LocalDateTime(2017, 10, 29, 1, 45, 0);
+
+            DateTimeZone dublin = DateTimeZoneProviders.Tzdb["Europe/Dublin"];
+
+            ZonedDateTime beforeTransition = new ZonedDateTime(dt, dublin, Offset.FromHours(1));
+
+            var result = Snippet.For(beforeTransition.Plus(Duration.FromHours(1)));
+
+            Assert.AreEqual(new LocalDate(2017, 10, 29), result.Date);
+            Assert.AreEqual(new LocalTime(1, 45, 0), result.TimeOfDay);
+            Assert.AreEqual(Offset.FromHours(0), result.Offset);
+
+            ZonedDateTime afterTransition = new ZonedDateTime(dt, dublin, Offset.FromHours(0));
+            result = afterTransition.Plus(Duration.FromDays(1));
+            Assert.AreEqual(new LocalTime(1, 45, 0), result.TimeOfDay);
+            Assert.AreEqual(new LocalDate(2017, 10, 30), result.Date);
+        }
+
+        [Test]
+        public void PlusHoursMinutesEtc()
+        {
+            DateTimeZone dublin = DateTimeZoneProviders.Tzdb["Europe/Dublin"];
+            var start = Instant.FromUtc(2017, 7, 17, 7, 17);
+            ZonedDateTime subject = new ZonedDateTime(start, dublin);
+            Assert.AreEqual(new ZonedDateTime(start + Duration.FromNanoseconds(1), dublin),
```

Comment:
I think it might be better to split these snippets up, one per unit. (I know that's annoying, but I think it'll give better results.)

What's slightly worrying is that the string representation may not go down to nanoseconds by default, so possibly the best format would be:

- Create the subject
- Create the result by adding the unit
- Assert a specific string result by formatting with the `F` standard pattern (which does down to the nanosecond, contrary to the documentation which I'm about to fix...)